ultra/ultra/train.py

```diff
@@ -134,20 +134,24 @@ def load_agents(experiment_dir):
         agent_id: agent_metadata["agent_classes"][agent_id] for agent_id in agent_ids
     }
 
-    agent_checkpoint_directories = {
-        agent_id: sorted(
-            glob.glob(os.path.join(experiment_dir, "models", agent_id, "*")),
-            key=lambda x: int(x.split("/")[-1]),
-        )
-        for agent_id in agent_ids
-    }
+    if len(os.listdir(os.path.join(experiment_dir, "models"))) != 0:
+        latest_checkpoint_directories = {
+            agent_id: sorted(
+                glob.glob(os.path.join(experiment_dir, "models", agent_id, "*")),
+                key=lambda x: int(x.split("/")[-1]),
+            )[-1]
+            for agent_id in agent_ids
+        }
+    else:
+        print("No models found, building agent_spec without any checkpoint directories")
+        latest_checkpoint_directories = {agent_id: None for agent_id in agent_ids}
```

Comment:
I know if one agent has a checkpoint directory, then all other agents should also have a checkpoint directory (because evaluation is done at the same episodic rate for all agents). However, for robustness sake and to be consistent, perhaps we can check, for each agent, whether they have a checkpoint directory:
```suggestion
    latest_checkpoint_directories = {}
    for agent_id in agent_ids:
        checkpoint_directories = sorted(
            glob.glob(os.path.join(experiment_dir, "models", agent_id, "*")),
            key=lambda x: int(x.split("/")[-1]),
        )
        latest_checkpoint_directory = checkpoint_directories[-1] if len(checkpoint_directories) > 0 else None
        latest_checkpoint_directories[agent_id] = latest_checkpoint_directory
```
I haven't tested this yet, but did you have a test that you ran when you originally caught this error?

Reply:
Changes applied